### ISSUE TITLE
Package request: foliate on darwin

### ISSUE THREAD

**Issue by sund3RRR** [NixOS/nixpkgs#302294]
**Project description**
As I understand, the main problem to build [Foliate](https://search.nixos.org/packages?channel=unstable&show=foliate&from=0&size=50&sort=relevance&type=packages&query=foliate) on macOS is gjs and webkitgtk binaries (look [here](https://github.com/johnfactotum/foliate/issues/115#issuecomment-509880889))
Both [gjs](https://search.nixos.org/packages?channel=unstable&show=gjs&from=0&size=50&sort=relevance&type=packages&query=gjs) and [webkitgtk](https://search.nixos.org/packages?channel=unstable&show=webkitgtk_6_0&from=0&size=50&sort=relevance&type=packages&query=webkitgtk) are presented in nixpkgs for darwin. I've tried to install Foliate using
```bash
nix-env -iA nixpkgs.foliate
```
but got
```
error: builder for '/nix/store/wqd6mh76yg9k29jjag39j7q33x7ygm17-webkitgtk-2.44.0+abi=6.0.drv' failed with exit code 1;
       > -- Could NOT find LibDRM (missing: LIBDRM_LIBRARIES LIBDRM_INCLUDE_DIR)
       > CMake Error at Source/cmake/OptionsGTK.cmake:316 (message):
       >   libdrm is required for USE_LIBDRM
       > Call Stack (most recent call first):
       >   Source/cmake/WebKitCommon.cmake:237 (include)
       >   CMakeLists.txt:21 (include)
```
I also cloned fresh nixpkgs repo and tried to build it manually, but got
```
      error: Package ‘spidermonkey-115.2.0’ in /Users/sunder/dev/nixpkgs/pkgs/development/interpreters/spidermonkey/common.nix:191 is marked as broken, refusing to evaluate.

       a) To temporarily allow broken packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_BROKEN=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowBroken = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowBroken = true; }
       to ~/.config/nixpkgs/config.nix.
```
After bypass this error by setting `export NIXPKGS_ALLOW_BROKEN=1`, I got the libdrm error again
```
-- Could NOT find LibDRM (missing: LIBDRM_LIBRARIES LIBDRM_INCLUDE_DIR) 
CMake Error at Source/cmake/OptionsGTK.cmake:316 (message):
  libdrm is required for USE_LIBDRM
Call Stack (most recent call first):
  Source/cmake/WebKitCommon.cmake:237 (include)
  CMakeLists.txt:21 (include)
```

I don't quite understand why webkitgtk is marked as supported on darwin, if the build fails with libdrm error.
But I think that these problems can potentially be resolved and we can get Foliate running on macOS.
Unfortunately I don't know how to fix this as I lack knowledge about building nix packages on macOS.
If someone takes up this matter, I would be very grateful, because Foliate is a wonderful program.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by skykanin** [comment]
Webkitgtk used to build on darwin, but doesn't anymore which is why it supports darwin, but it also marked as broken. See https://github.com/NixOS/nixpkgs/pull/145115

TL;DR: Fixing webkitgtk for darwin probably requires understanding of the code to fix the compilation issues
